packages/experimental/Tooltip/src/Tooltip.tsx

```diff
@@ -9,21 +10,33 @@ import { tooltipName } from './Tooltip.types';
 
 const NativeTooltipView = ensureNativeComponent('RCTTooltip');
 
-/**
- * A function which determines if a set of styles should be applied to the component given the current state and props of the tooltip.
- *
- * @param layer The name of the state that is being checked for
- * @param userProps The props that were passed into the tooltip
- * @returns Whether the styles that are assigned to the layer should be applied to the tooltip
- */
-export const tooltipLookup = (layer: string, userProps: TooltipProps): boolean => {
-  return userProps[layer];
-};
-
 export const Tooltip = stagedComponent((props: TooltipProps) => {
-  return (rest: TooltipProps, children: React.ReactNode) => {
-    return <NativeTooltipView {...mergeProps(props, rest)}>{children}</NativeTooltipView>;
+  const { target } = props;
+
+  const [nativeTarget, setNativeTarget] = React.useState<number | string | null>(null);
+
+  React.useLayoutEffect(() => {
+    if (typeof target === 'string') {
+      // Pass string type `target` directly
+      setNativeTarget(target);
+    } else if (target?.current) {
+      // Pass the tagID for a valid ref `target`
+      setNativeTarget(findNodeHandle(target.current));
```

Comment:
I know this is existing code and used elsewhere, but it's worth noting `findNodeHandle`, while it works in the new arch, isn't recommended and can break. I think that's why the link I sent earlier uses a Rect instead of a ref to position its' tooltip. It might be worth discussing whether we just go with using a Rect altogether

Reply:
Thanks for sharing the link to the new architecture! I'll make a note to look into this / discuss this when we revisit Tooltip in the future.